se/se_epub_generate_toc.py

```diff
@@ -412,17 +456,17 @@ def process_headings(soup: BeautifulSoup, textf: str, toc_list: list, nest_under
 		special_item = TocItem()
 		# Need to determine level depth.
 		# We don't have a heading, so get first content item
-		content_item = soup.find(["p", "header", "img"])
+		content_item = dom.xpath("//p | //header | //img")
 		if content_item is not None:
-			parents = content_item.find_parents(["section", "article"])
+			parents = get_outer_sections(content_item[0], ["section", "article"])
 			special_item.level = len(parents)
 			if special_item.level == 0:
 				special_item.level = 1
 		if nest_under_halftitle:
 			special_item.level += 1
-		title_tag = soup.find("title")  # Use the page title as the ToC entry title.
+		title_tag = dom.xpath("//head/title")  # Use the page title as the ToC entry title.
```

Comment:
This can also be simplified, the xpath should be: `dom.xpath("//head/title/text()", True)`.

Because we passed True to the xpath function, this will either return a string containing the text of the title tag, or None if there is no match. Then the if/then logic below this should be unnecessary.